### ISSUE TITLE
[Bug]: 4.48.0 caused permanent diffs with stateful_rule rule_options between keyword/settings and keyword mapping

### ISSUE THREAD

**Issue by realpdm** [hashicorp/terraform-provider-aws#28649]
### Terraform Core Version

1.1.9

### AWS Provider Version

4.48.0

### Affected Resource(s)

When using the aws_networkfirewall_rule_group resource and specifying a suricata "sid" value as the rule_options terraform will send them as separate keyword and setting values. When it compares the state against what is configured in AWS AWS responds with the keyword and setting all in the 'keyword' field while setting field is null. 

This started on Dec 20, the day after 4.48.0 was released with no other changes to the terraform code. 

### Expected Behavior

A plan with no changes to apply should not return any changes to be made. 

### Actual Behavior

What terraform is sending to AWS does not match what AWS seems to be configuring. 
Example:
```
    ~ stateful_rule {
                    # (1 unchanged attribute hidden)


                  + rule_option {
                      + keyword  = "sid"
                      + settings = [
                          + "110137",
                        ]
                    }
                  - rule_option {
                      - keyword  = "sid:110137" -> null
                      - settings = [] -> null
                    }
                    # (1 unchanged block hidden)
```               

### Relevant Error/Panic Output Snippet

```shell
No error, just repeated diff to apply
```

### Terraform Configuration Files

```
resource "aws_networkfirewall_rule_group" "access_list" { 
...normal rule stuff...
          rule_option {
            keyword = "sid"
            settings = [
              var.rule_groups[each.key].sid_base + index(each.value.access_list, stateful_rule.value)
            ]
          }
}
```
this problem also happens if the format for the rule_option is the shortcut style:

```
          rule_option {
            keyword = "sid:12345"
          }
```

When the provider maps this to AWS settings it seems to split the keyword/setting on the colon in the string. 

### Steps to Reproduce

When I run plan again with no changes, the mentioned above is shown for every rule. 

### Important Factoids

I have drift detection running and the diff started happening on December 20, 2022, which is the day after 4.48.0 was released. There were no changes to the terraform code. 

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
Likely relates: https://github.com/hashicorp/terraform-provider-aws/pull/27102.

**Comment by realpdm** [comment]
I think seems to have been fixed in 4.49.0 though I didn't see it mentioned in the Changelog

This looks like a likely fix but I'm not familiar enough with the code to be sure. 

https://github.com/hashicorp/terraform-provider-aws/commit/7baa7c8a4bb629b1f0daa6533bfc61df69f4c170

At any rate, seems resolved.
